remove untrue/contentious statement

tshepang opened the pull request:
One can use unstable features on stable too, like via an env var

r? ghost

jieyouxu commented:
The current statement I find is accurate, and less confusing than trying to caveat it with "intended".

BoxyUwU commented:
RUSTC_BOOTSTRAP is very much not widely used

tshepang commented:
I should of chosen a different word

jieyouxu commented:
I actually find removing the intended usage to be more confusing, so I think we should retain current wording.

tshepang commented:
I see now how I made it less helpful. How about now...
- I restored the explanation
- I removed the wrong claim
- I added the intent (to imply that people should avoid using feature gates in beta and stable)
